src/garage/experiment/local_tf_runner.py

```diff
@@ -319,6 +324,57 @@ def train(self,
             pause_for_plot=pause_for_plot,
             start_epoch=0)
 
+        self.plot = plot
+
+        return self.algo.train(self, batch_size)
+
+    def step_epochs(self):
+        """Generator for training.
+
+        This function serves as a generator. It is used to separate
+        services such as snapshotting, sampler control from the actual
+        training loop. It is used inside train() in each algorithm.
+
+        The generator initializes two variables: `self.step_itr` and
+        `self.step_path`. To use the generator, these two have to be
+        updated manually in each epoch, as the example shows below.
+
+        Yields:
+            int: The next training epoch.
+
+        Examples:
+            for epoch in runner.step_epochs():
+                runner.step_path = runner.obtain_samples(...)
+                self.train_once(...)
+                runner.step_itr += 1
+
+        """
+        try:
+            self.start_worker()
+
+            self.start_time = time.time()
+
+            self.step_itr = (
+                self.train_args.start_epoch * self.train_args.n_epoch_cycles)
+            self.step_path = None
+
+            for epoch in range(self.train_args.start_epoch,
+                               self.train_args.n_epochs):
+                self.itr_start_time = time.time()
+
+                with logger.prefix('epoch #%d | ' % epoch):
+                    yield epoch
+
+                    save_path = (self.step_path
+                                 if self.train_args.store_paths else None)
+                    self.save(epoch, save_path)
+
+                    self.log_diagnostics(self.train_args.pause_for_plot)
+                    logger.dump_all(self.step_itr)
+                    tabular.clear()
+        finally:
+            self.shutdown_worker()
```

Comment:
unless they are called from outside this class, please make these methods _private

Reply:
We have some tests using the `start_worker()` and `shutdown_worker()` in `LocalRunner`, but not in garage. Is it good practice to make these methods private?

Reply:
tests may access private members if necessary, but they should be treated as private within the non-test code (so it sounds like these should be private)